data-prepper-plugins/common/src/main/java/org/opensearch/dataprepper/plugins/sink/FileSink.java

```diff
@@ -77,14 +78,19 @@ public void output(final Collection<Record<Object>> records) {
                 return;
 
             for (final Record<Object> record : records) {
+                if (record.getData() instanceof Event) {
+                    Event event = (Event)record.getData();
+                    sinkForwardRecordsContext.addRecord(new Record<>(event));
```

Comment:
Why do you create a new `Record`?

Reply:
@dlvenable because FileSink uses `Record<Object>` and not `Record<Event>`. It looks like I can't simply cast it. I don't think we should change `forwardRecords` to accept `Record<Object>`